dirty_cat/minhash_encoder.py

```diff
@@ -184,14 +195,48 @@ def transform(self, X):
             X = X[:, 0]
         # Check if first item has str or np.str_ type
         assert isinstance(X[0], str), "ERROR: Input data is not string."
+        if self.minmax_hash:
+            assert self.n_components % 2 == 0,\
+                    "n_components should be even when minmax_hash=True"
+        if self.hashing == 'murmur':
+            assert not(self.minmax_hash),\
+                   "minmax_hash not implemented with murmur"
+        if self.handle_missing not in ['error', '']:
+            template = ("handle_missing should be either 'error' or "
+                        "'', got %s")
+            raise ValueError(template % self.handle_missing)
         X_out = np.zeros((len(X), self.n_components))
+        X = X.ravel()
```

Comment:
This ravel is new, isn't it?

Why do we need it? I wonder if it won't have side effects.

Reply:
The subsequent for loop check breaks if we have an array shape of (N, 1). Ravel turns it into a 1-D array. I can use np.resize instead.

Reply:
OK. I worry about data being put in with a shape of (N, P), where the ravel could have "interesting" consequences. Maybe this cannot happen, and we can worry about it later, if we hit the problem.

Reply:
So, we have this on line 175 to assert a shape of (N, ) or (N, 1). At this time a resize later on, can assume this shape, in which case I think it should not be a problem? I used resize() instead in case ravel() is newer. 

```
assert X.ndim == 1 or (X.ndim == 2 and X.shape[1] == 1), f"ERROR:\
        shape {X.shape} of input array is not supported."
```